src/temporal_terms.jl

```diff
@@ -0,0 +1,60 @@
+# Basic Lead/Lag:
+
+# syntax:
+lead_lag_docstring(op, opped, prevnext, firstlast) = """\n
+        $op(term, nsteps::Integer)
+
+    This `@formula` term is used to introduce $opped variables.
+    For example `$op(x,1)` effectively adds a new column containing
+    the value of the `x` column from the $prevnext row.
+    If there is no such row (e.g. because this is the $firstlast row),
+    then the $opped column will contain `missing` for that entry.
+
+    Note: this is only a basic row-wise $op operation.
+    It is up to the user to ensure that data is sorted by the temporal variable,
+    and that observations are spaced with regular time-steps.
+    (Which may require adding extra-rows filled with `missing` values.)
+"""
+
+@doc lead_lag_docstring("lag", "lagged", "previous", "first") lag
+@doc (lead_lag_docstring("lead", "lead", "next", "last")) lead
+
+
+# struct for behavior
+struct LeadLagTerm{T<:AbstractTerm, F<:Union{typeof(lead), typeof(lag)}} <: AbstractTerm
+    term::T
+    nsteps::Int
+end
+
+op(::LeadLagTerm{<:Any, typeof(lag)}) = lag
```

Comment:
Why not use `F.instance` as in `apply_schema`?

Reply:
In an earlier version of this code `op` was called in a bunch of places,.
but not it is only called in 1 so that makes sense.